yt/yt/library/containers/cri/cri_executor.cpp

```diff
@@ -544,6 +544,25 @@ class TCriExecutor
                 }));
         }
 
+        auto pullComplete = NewPromise<void>();
+        {
+            auto guard = Guard(SpinLock_);
+            if (auto* pullFuture = InflightImagePulls_.FindPtr(image.Image)) {
+                return pullFuture->Apply(BIND([=, this, this_ = MakeStrong(this)] {
+                    return PullImage(image, /*always*/ false, authConfig, podSpec);
+                }));
+            }
+
+            auto pullFuture = pullComplete.ToFuture().ToUncancelable();
+            EmplaceOrCrash(InflightImagePulls_, image.Image, pullFuture);
+            pullFuture.Subscribe(BIND([=, this, weakThis = MakeWeak(this)] (const TError &) {
```

Comment:
Here `const TError&` is not required too

Reply:
I prefer to leave it here to express that it is called even in case of error.

Reply:
Let's log the error in this case. And `const TError &` -> `const TError&`. And let's do the same for the snippet above.

Reply:
This seems redundant. I've added logging of original docker pull error. Chained pulls will log their progress after that, and all messages linked by same image name.